internal/pkg/gateway/registry.go

```diff
@@ -205,21 +205,31 @@ func (reg *registry) evaluator(channel string, chaincode string, targetOrgs []st
 			targetOrgs = append(targetOrgs, org)
 		}
 	}
-	// Prefer a local org endorser, if present
-	if e, ok := endorsersByOrg[reg.localEndorser.mspid]; ok && contains(targetOrgs, reg.localEndorser.mspid) {
-		return e[0].endorser, nil
-	}
-	// Otherwise highest block height peer (first in list) from another org
-	var evaluator *endorser
-	var maxHeight uint64
+
+	localOrgEndorsers := []*endorserState{}
+	otherOrgEndorsers := []*endorserState{}
 	for _, org := range targetOrgs {
-		if e, ok := endorsersByOrg[org]; ok && e[0].height > maxHeight {
-			evaluator = e[0].endorser
-			maxHeight = e[0].height
+		if es, ok := endorsersByOrg[org]; ok {
+			for _, e := range es {
+				if org == reg.localEndorser.mspid {
+					localOrgEndorsers = append(localOrgEndorsers, e)
+				} else {
+					otherOrgEndorsers = append(otherOrgEndorsers, e)
+				}
+			}
```

Comment:
For loop can be avoided

```suggestion
			if org == reg.localEndorser.mspid {
				localOrgEndorsers = es
			}else{
				otherOrgEndorsers = append(otherOrgEndorsers, es...)
			}
```